docs/Project-Configuration/Project-Commands.md

```diff
@@ -935,7 +936,12 @@ Parameters:
     1.0.1).
 -   `filenames`: A list of filename paths to pass to the service. You may use
     full filepaths in this parameter, the command will label the file with its
-    basename only when sent to the service.
+    basename only when sent to the service. Wildcard globs are supported within
+    a single directory path. For example, the filename `dist/*.zip` would
+    locate each zip file within the `dist` directory and individually trace
+    those files. Double star globs like `dist/**/*.zip` are not supported. If
+    a filename is matched multiple times in the same call to `papertrail.trace`,
```

Comment:
Is there a reason we can't just de-duplicate the duplicate matching files? This seems like it could be inconvenient to use if you have to ensure every pattern has no overlap.

Reply:
I hit submit before seeing your review 🙈

Reply:
The main reason I opted for this approach is that Papertrail thinks in terms of basenames, but users will pass files in terms of file paths. This means that it's hard for us to know which pattern to pick. e.g. in the following situation:

```
.
..
./builda/file.tar.gz
./buildb/file.tar.gz
```

The user will want to trace `file.tar.gz` twice, even though those filenames refer to different files. We could de-duplicate based on filename and checksum, but I think it's safer to let users know there's a conflict and that they should reconsider their setup. I'd rather be conservative on this now and then make it accept more later on if this proves annoying. I already ran into a setup somewhat at risk of this when working with projects, which is why I added the protection.

Reply:
Gotcha, that makes sense. I still think it would help to clarify what this means a little bit, since it's unexpected behavior and may cause some user confusion. A typical user is likely to think they're different file paths in the file system, so therefore Papertrail should accept both. Can we clarify this a little bit for users?

* In the docs where it discusses the basename behavior, it would help to provide an example showing that two files with the same basename in different paths is invalid. The example you gave is pretty illuminating, so I think users will pick up on it pretty easily once they see that example.
* In the error message itself, can we clarify that it's not a bug but rather a legitimate issue with the user input? It should be sufficient to just include an extra sentence with the duplicate file error saying "Papertrail requires file names to be unique regardless of their path".